### ISSUE TITLE
Highlights misaligned with bold text

### ISSUE THREAD

**Issue by trishume** [google/xi-mac#50]
Background-color highlights like are used for find matches and selection and the source text don't align properly in the presence of bold text. I assume because the rendering takes into account the style but the measurement doesn't.

I discovered this while playing around with https://github.com/google/xi-editor/pull/335 and noticing that the highlights weren't aligned correctly in Rust function definitions which are bolded. Notice in the screenshot below the "self" highlight in `&mut self` is aligned without taking into account the wider bold characters.

![screen shot 2017-06-06 at 2 44 27 pm](https://user-images.githubusercontent.com/887610/26853535-47417d22-4ac7-11e7-9e68-a0db656bba19.png)

**Comment by raphlinus** [comment]
Good catch. Most of us are using monospaced font, where this wouldn't come up, but it should be fixed for the general case.

I think the ultimate solution to this is to store attributed text lines (maybe CTLine) in the line cache, not json'ish data structures. Then when we do measurement from those, it should always line up with the drawing. Extra bonus, should be a considerable performance improvement.
